common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -1009,9 +1013,16 @@ def student_view_data(self, context=None):
                 }
 
         transcripts_info = self.get_transcripts_info()
+        available_translations = self.available_translations(transcripts_info)
+        available_translations += get_available_transcript_languages(
+            course_id=self.runtime.course_id.for_branch(None),
+            edx_video_id=self.edx_video_id,
+            youtube_id_1_0=self.youtube_id_1_0,
+            html5_sources=self.html5_sources,
+        )
         transcripts = {
             lang: self.runtime.handler_url(self, 'transcript', 'download', query="lang=" + lang, thirdparty=True)
-            for lang in self.available_translations(transcripts_info)
+            for lang in set(available_translations)
```

Comment:
Why `set` again here?

Reply:
`get_available_transcript_languages` may returns codes that are already in `available_translations`